index.ts

```diff
@@ -593,102 +691,71 @@ commander
-                    name: formattedName, // the file base name
-                    type: getAPIFileType(name), // the file extension
-                    source: contents[i] //the file contents
-                  };
-                  return file;
-                } else {
-                  return; // Skip ignored files
-                }
-              }).filter(Boolean); // remove null values
-
-              script.projects.updateContent({
-                scriptId,
-                resource: { files }
-              }, {}, (error: any, res: Function) => {
-                spinner.stop(true);
-                if (error) {
-                  console.error(LOG.PUSH_FAILURE);
-                  error.errors.map((err: any) => {
-                    console.error(err.message);
-                  });
-                  console.error(LOG.FILES_TO_PUSH);
-                  nonIgnoredFilePaths.map((filePath) => {
-                    console.error(`└─ ${filePath}`);
-                  });
-                } else {
-                  nonIgnoredFilePaths.map((filePath) => {
-                    console.log(`└─ ${filePath}`);
-                  });
-                  console.log(LOG.PUSH_SUCCESS(nonIgnoredFilePaths.length));
-                }
-              });
-            });
+        getProjectFiles(rootDir, (err, projectFiles, files) => {
+          if (err !== undefined) {
+            return;
+          } else if (projectFiles !== undefined) {
+            const [nonIgnoredFilePaths, ignoredFilePaths] = projectFiles;
+            script.projects.updateContent({
+              scriptId,
+              resource: { files }
+            }, {}, (error: any, res: Function) => {
+              spinner.stop(true);
+              if (error) {
+                console.error(LOG.PUSH_FAILURE);
+                error.errors.map((err: any) => {
+                  console.error(err.message);
+                });
+                console.error(LOG.FILES_TO_PUSH);
+                nonIgnoredFilePaths.map((filePath) => {
+                  console.error(`└─ ${filePath}`);
+                });
+              } else {
+                nonIgnoredFilePaths.map((filePath) => {
+                  console.log(`└─ ${filePath}`);
+                });
+                console.log(LOG.PUSH_SUCCESS(nonIgnoredFilePaths.length));
+              }
           });
-        });
+        }
       });
     });
   });
+});
+
+/**
+ * Lists files that will be written to the server on `push`.
+ * Ignores files:
+ * - That start with a .
+ * - That don't have an accepted file extension
+ * - That are ignored (filename matches a glob pattern in the ignore file)
+ */
+commander
+.command('status')
+.description('Lists files that will be pushed by clasp')
+.action(async () => {
+  getProjectSettings().then(({ scriptId, rootDir }: ProjectSettings) => {
+    if (!scriptId) return;
+
+    getProjectFiles(rootDir, (err, projectFiles) => {
+      if (err !== undefined) {
```

Comment:
Maybe add types to the method params to make the warnings go away.

Ideally, we should condense the code this way.
```ts
if (err) return console.error(err);
const [nonIgnoredFilePaths, ignoredFilePaths] = projectFiles;
...
```

Reply:
ACK